gitops/teamcity/README.md

```diff
@@ -0,0 +1,92 @@
+
+    ![](./images/output_available.png)
+
+9. When you visit this URL, for example `http://server-teamcity-999608bafa.eastus2.cloudapp.azure.com/showAgreement.html`, it will display the licence agreement for JetBrains TeamCity. Press `Accept` and `Continue`. 
+10. Next, it will prompt you to create a new administrator account. Enter a username and secure password, then press `Create Account`. 
+
+    ![](./images/create_admin_account.png)
+
+11. Next, it will prompt you to fill in your profile. Complete your profile and navigate to `Projects` in the top left. It should display an option to Create a project. 
+
+    ![](./images/create_project.png)
+
+12. Create the project from the repository URL, and enter credentials if it's not a public repository. Click on `Proceed`. 
+    
+    ![](./images/from_repository.png)
+
+13. Enter a name for the build and click on `Proceed`. 
+    
+    ![](./images/build_name.png)
+
+14. After this step, it will auto-detect build steps and prompt you to configure it, and if none are present you may configure it manually. Click on `configure build steps manually`, since we're planning on using the build script provided by  [`Microsoft` / `Bedrock`](https://github.com/microsoft/bedrock). 
+15. Select `Command Line` from the dropdown for `Runner Type`. Enter a step name and select `Custom script` for `Run`. The script we're using below downloads the Bedrock GitOps build script, which downloads all essential tools needed to convert high level definition files into Kubernetes manifest yaml files, and pushes them into a manifest repo. All you need to use are the following simple commands for `Custom script`: 
+    ```
+    curl https://raw.githubusercontent.com/microsoft/bedrock/master/gitops/teamcity/build.sh > build.sh
+    chmod +x ./build.sh
+    ./build.sh
+    ```
+    ![](./images/transform_and_publish_step.png)
+
+16. Press `Save` and click on `Parameters` on the left column. 
+    ![](./images/parameters_find.png)
+
+17. Click on `Add new parameter`, and add the following variables under `Environment Variables`:
+    - `ACCESS_TOKEN_SECRET`: Get an access token from GitHub/Azure DevOps where your repo is hosted, so that the script can have access to push to the remote manifest yaml repository. You should use the password type for this variable for security purposes (as shown below). 
+        ![](./images/password_variable.png)
+    - `COMMIT_MESSAGE`: Since TeamCity does not have a variable that can provide the exact commit message, enter something valuable, such as "Auto-generated yaml files"
+    - `REPO`: Set this to the manifest repo, such as `https://github.com/samiyaakhtar/jackson-manifest`
+        ![](./images/variables.png)
+
+18. Click on `Run` to do a test run for this configuration! 
+
+You should be able to see logs from the build that just completed, and in the end it should push manifest files to the repository.
+
+![](./images/build_complete.png)
+
+## Disadvantages
```

Comment:
To be less cynical perhaps we frame the title to be "Challenges with TeamCity"? What do you think?

Reply:
Yup that sounds good :)